docs/faqs.md

```diff
@@ -340,6 +340,12 @@ Keep reading in the [consuming recipes section](consuming_recipes.md).
 
 ## Why are version ranges not allowed?
 
-Version ranges are a useful Conan feature, find the documentation [here](https://docs.conan.io/en/latest/versioning/version_ranges.html). However, in the context of ConanCenter they pose a few key challenges, most notably "Build Reproducibility".
+Version ranges are a useful Conan feature, find the documentation [here](https://docs.conan.io/en/latest/versioning/version_ranges.html). However, in the context of ConanCenter they pose a few key challenges, most notably:
+
+- Non-Determinstic `package-id`
+
+With version ranges the newest compatible package may yeild a different package-id then the one built and published by ConanCenter resulting in fustrating error "no binaries found". For more context see [this excellent explanation](https://github.com/conan-io/conan-center-index/pull/8831#issuecomment-1024526780).
+
+- Build Reproducibility
 
 If consumers try to download and build the recipe at a later time, it may resolve to a different package version that may not be compatible. In order to prevent these types of issues, we have decided to only allow exact requirements versions.
```

Comment:
This statement is strange. Version range is tracks compatible versions, so it's the opposite, it would help to avoid users specifying incompatible versions.

Reply:
🤔 Should this be removed? as it was pointed out, even with exact versions there's no guarantee

Reply:
not every arbitrary version range expression describes compatible versions, e.g. `[1.0 || 2.0]` doen't

Reply:
Talking about _build reproducibility_ is not the same as talking about compatibility. Different versions of a requirement can be compatible, but for sure if we are using different ones for different builds, we cannot say we are _reproducing_ the same build.

Anyway, unless we use lockfiles to ensure that we are using at least the same recipe-revision we cannot claim anything about build-reproducibility... or at least as I understand it, closer to _deterministics builds_ concept ([blogpost](https://blog.conan.io/2019/09/02/Deterministic-builds-with-C-C++.html)).

Here the reason is more about **maximizing the chances of success when building a recipe again**: from time to time in CCI we need to build again some recipe because we have added more configurations, or users submit a PR to modify things not related to requirements... in both cases, the chances that the PR will succeed are greater if we keep using the same version of the requirements.

Reply:
Would be great if we could have a tradeoff:
- fixed version
- information about compatible versions of direct dependencies for each version of a recipe.

Reply:
I love that idea.

It would be great to have a feature in Conan to declare which versions are compatible, even though the requirement is fixed. In some issues, we thought about adding this information to comment just after the statement, or to the `config.yml` file... so the CI can open PRs to bump dependencies whenever new compatible versions are available. Having it as a Conan feature would be much better, for sure.

Reply:
For sure it should be declarative (in `conandata.yml`, not the `config.yml` which is not part of the recipe).
It may not be straightfoward due to conditional requirements. I guess it should accept to declare compatible versions of a dependency even if it's not part of the dependency graph after `requirements()` resolution.

Reply:
The only problem I can see is that compatibility range might be different for different configurations. It's not only that requirements are conditional (let's list all of them 👍 ). I would agree (from bot perspective) to list only the intersection of all the compatible ones, at least in the context of CCI, outside some people might find this too restrictive (that's why maybe `config.yml` which is local to CCI is a better place to start trying this feature)